### PR TITLE
Turn generator into a lib, aio-cli-lib-console

### PR DESCRIPTION
**Pull request by moritzraho**
## Description
- Requires changes in app plugin and console plugin for the require path
- Requires https://github.com/adobe/aio-cli-plugin-app/pull/398 to get rid of the actual generator deleted here

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by shazron — APPROVED** [review]
some merge issues that need resolving, but LGTM

**Comment by moritzraho** [comment]
thanks Shaz, resolved merge issues, will merge this after tests run

**Pull request merged**
